webpack/assets/javascripts/foreman_store.js

```diff
@@ -0,0 +1,34 @@
+import { get } from 'lodash';
+import store from './react_app/redux';
+
+/**
+ * Observe a state from the store for changes
+ * @param  {String}   state a sub object of the store
+ * @param  {Function} onChange  a callback to run when the store has been changed
+ * @return {Function}           unsubscribe function
+ */
+export function observeStore(state, onChange) {
+  let currentState;
+
+  function handleChange() {
+    const nextState = select(state);
+    if (nextState !== currentState) {
+      currentState = nextState;
+      onChange(currentState, unsubscribe);
+    }
+  }
+
+  const unsubscribe = store.subscribe(handleChange);
+  handleChange();
+  return unsubscribe;
+}
+
+function select(state) {
+  const stateObj = get(store.getState(), state);
```

Comment:
Shouldn't you receive the `nextState` from the `store.subscribe` method instead of calling `store.getState()`?

Reply:
AFAIK, `store.subscribe` doesn't have  `nextState` as a param